File: src/main/java/com/xudong/vam/service/impl/SelectServiceImpl.java
package com.xudong.vam.service.impl;

import com.xudong.vam.model.SelectPackage;
import com.xudong.vam.repository.SelectDetailRepository;
import com.xudong.vam.repository.SelectPackageRepository;
import com.xudong.vam.service.SelectService;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
@AllArgsConstructor
public class SelectServiceImpl implements SelectService {
    private final SelectPackageRepository selectPackageRepository;

    private final SelectDetailRepository selectDetailRepository;

    @Override
    public long createSelect(String name) {
        SelectPackage selectPackage = selectPackageRepository.findByName(name);
        if (selectPackage != null) {
            return selectPackage.getId();
        }

        selectPackage = new SelectPackage(name, UUID.randomUUID().toString());
        selectPackage = selectPackageRepository.save(selectPackage);

        return selectPackage.getId();
    }
}

File: src/main/java/com/xudong/vam/core/service/impl/SelectServiceImpl.kt
package com.xudong.vam.core.service.impl

import com.xudong.vam.core.model.SelectPackage
import com.xudong.vam.core.repository.SelectPackageRepository
import com.xudong.vam.core.service.SelectService
import lombok.AllArgsConstructor
import org.springframework.stereotype.Service
import java.util.*

@Service
@AllArgsConstructor
class SelectServiceImpl : SelectService {
    private val selectPackageRepository: SelectPackageRepository? = null

    override fun createSelect(name: String): Long {
        var selectPackage = selectPackageRepository!!.findByName(name)
        if (selectPackage != null) {
            return selectPackage.id!!
        }

        selectPackage = SelectPackage(name = name, uuid = UUID.randomUUID().toString())
        selectPackage = selectPackageRepository.save(selectPackage)

        return selectPackage.id!!
    }
}
